internal/controller/postgrescluster/pgmonitor.go

```diff
@@ -97,10 +99,14 @@ func (r *Reconciler) reconcilePGMonitorExporter(ctx context.Context,
 
 		for _, containerStatus := range writablePod.Status.ContainerStatuses {
 			if containerStatus.Name == naming.ContainerPGMonitorExporter {
-				setup = containerStatus.ImageID
+				exporterImageSHA = containerStatus.ImageID
+			}
+			if containerStatus.Name == naming.ContainerDatabase {
+				pgImageSHA = containerStatus.ImageID
 			}
 		}
-		if setup == "" {
+
+		if exporterImageSHA == "" {
```

Comment:
🤔 `pgImageSHA` goes into the hash as well. Should proceed only when we have both (i.e. return early when we lack either)?

⛏️ It might be good to explain what's going on in this block and/or why; around L90, maybe.

Reply:
My thought process on not checking that was that if we don't have the pg image, it wouldn't be a writable pod, but now thinking about it, I like the idea of the explicit check. Will change that and add an explanation.